File: WebFt30/src/app/core/store/models/system-event.ts
import { SystemEventLevel } from './system-event-level';
import { SystemEventSource } from './system-event-source';

export class SystemEvent {
  id!: number;
  type!: string;
  level!: SystemEventLevel;
  jsonData!: string;
  source!: SystemEventSource;
  at!: Date;
}

File: Backend/src/Fibertest30.Application/SystemEvent/Models/SystemEvent.cs
namespace Fibertest30.Application;

public class SystemEvent : INotificationEvent
{
    public int Id { get; set; } = 0!;
    public SystemEventType Type { get; init; }
    public SystemEventLevel Level { get; init; } 
    public ISystemEventData? Data { get; init; }  // null if event has no data
    public SystemEventSource Source { get; init; }
    
    public DateTime At { get; set; }

    public SystemEvent(SystemEventType type, 
        SystemEventLevel level,
        ISystemEventData? data,
        SystemEventSource source)
    {
        Type = type;
        Level = level;
        Data = data;
        Source = source;
    }
}


